inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -802,9 +806,37 @@ void prepare_primitive_fusing::fuse_simple_primitives(program_impl &p) {
             if (parent2->is_type<convolution>() && !conv_supports_fusings(parent2->as<convolution>()))
                 return;
 
-            // This fusing can be extended to support peer node in any layout
-            bool merge_allowed = fused_node->get_users().size() == 1;
+            bool merge_allowed = true;
+            if (fused_node->is_type<convolution>() && fused_node->get_users().size() > 1) {
+                //std::pair<cldnn::program_node*, layer level>
+                std::deque<std::pair<cldnn::program_node*, int>> node_queue;
```

Comment:
why should this be deque? do we need to pop_back or push_front?

Reply:
Because we need iterator function of dequeue to remove duplication of node in deque.